dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/BitstreamRestRepository.java

```diff
@@ -135,6 +155,76 @@ protected void delete(Context context, UUID id) throws AuthorizeException {
         }
     }
 
+    /**
+     * Find the bitstream for the provided handle and sequence or filename.
+     * When a bitstream can be found with the sequence ID it will be returned if the user has "METADATA_READ" access.
+     *
+     * @param handle    The handle of the item
+     * @param sequence  The sequence ID of the bitstream
+     * @param filename  The filename of the bitstream
+     *
+     * @return a Page of BitstreamRest instance matching the user query
+     */
+    @SearchRestMethod(name = "byItemHandle")
+    public BitstreamRest findByItemHandle(@Parameter(value = "handle", required = true) String handle,
+                                          @Parameter(value = "sequence") Integer sequence,
+                                          @Parameter(value = "filename") String filename) {
+        if (StringUtils.isBlank(filename) && sequence == null) {
+            throw new IllegalArgumentException("The request should include a sequence or a filename");
+        }
+
```

Comment:
The easiest solution is to give precedence to the sequence since this is always unique on an item (put the `fileName` matching in `getMatchedBitstreams` to the else of `sequence!=null`). Additionally we could add the check if sequence match is found to also check the `fileName`, since the contract does not mention allowing non-matching sequence/fileName combinations.

Reply:
@MarieVerdonck : I agree that the sequence should take priority over filename as it's guaranteed unique.  So, if both are passed, use the sequence.  If somehow the sequence is invalid, then use the filename.